shell.nix

```diff
@@ -0,0 +1,6 @@
+{pkgs ? import <nixpkgs> {}}:
```

Comment:
Why do you need this file ?

Reply:
it is a file that list all packages needed to run/work on the project. this file is used by nix and some other tools that allow developers to have an easily reproducible dev environment. it is usually recommended to push it to allow other users to benefit from a pre-configured environment, but if you don't want it on master i can remove it.